### PR TITLE
Add GCP Chat with example

### PR DESCRIPTION
**Pull request by nomisRev**

### CODE REVIEW

**Review by raulraja — DISMISSED** [review]
It looks great @nomisRev !.

We need to also support the Embeddings wired to the GCP embeddings models. Currently, the `getOrThrow` imported in the example brings the embeddings from the GPT4All module which are actually not working great and depend on an adapter from dl java for HuggingFace.

I think the way we are using single embeddings generically for all documents in the `ai` while we have different models that may require tokenizing differently may not yield always the best results.

**Comment by nomisRev** [comment]
@raulraja do you want me to change anything in this PR, or shall we pick it up in subsequent PRs?

File: integrations/gcp/src/commonMain/kotlin/com/xebia/functional/xef/gcp/GcpChat.kt [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  private fun List<Message>.buildPrompt(): String {
+    val messages: String =
+      joinToString("") { message ->
+        when (message.role) {
+          Role.SYSTEM -> message.content
+          Role.USER -> "\n### Human: ${message.content}"
+          Role.ASSISTANT -> "\n### Response: ${message.content}"
+        }
+      }
+    return "$messages\n### Response:"
```
Comment (Montagon) [inline integrations/gcp/src/commonMain/kotlin/com/xebia/functional/xef/gcp/GcpChat.kt:107]: I'm thinking if this is mandatory or not 🤔 

From [here](https://cloud.google.com/vertex-ai/docs/generative-ai/chat/chat-prompts) we can see that the format should be something like this:
```json
"messages": [
  {
    "author": "USER",
    "content": "Hi",
  },
  { 
    "author": "AI",
    "content": "Hi! How can I help you today?",
  },
  { 
    "author": "USER",
    "content": "I just want to talk.",
  },
  { 
    "author": "AI",
    "content": "Sure, I'm here to listen. What would you like to talk about?",
  },
  { 
    "author": "USER",
    "content": "Let's talk about movies.",
  }],
  ```

File: integrations/gcp/src/commonMain/kotlin/com/xebia/functional/xef/gcp/GcpChat.kt [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import kotlinx.uuid.generateUUID
+
+@OptIn(ExperimentalStdlibApi::class)
+class GcpChat(apiEndpoint: String, projectId: String, modelId: String, token: String) :
+  Chat, Completion, AutoCloseable {
+  private val client: GcpClient = GcpClient(apiEndpoint, projectId, modelId, token)
+
+  override val name: String = client.modelId
+  override val modelType: ModelType =
+    ModelType.LocalModel(client.modelId, EncodingType.CL100K_BASE, 4096)
```
Comment (Montagon) [inline integrations/gcp/src/commonMain/kotlin/com/xebia/functional/xef/gcp/GcpChat.kt:25]: The model used in the example only has 2048 as max tokens. I don't know if is better to set here the value. I don't know if is possible to modify from another place: 
```suggestion
    ModelType.LocalModel(client.modelId, EncodingType.CL100K_BASE, 2048)
```

**Review by Montagon — APPROVED** [review]
Thanks Simon! Left only two comments

File: integrations/gcp/src/commonMain/kotlin/com/xebia/functional/xef/gcp/GcpChat.kt (hunk 1 above)
Comment (nomisRev) [reply integrations/gcp/src/commonMain/kotlin/com/xebia/functional/xef/gcp/GcpChat.kt:107]: Let’s revisit this in a later PR. Is that okay with you?

Comment (Montagon) [reply integrations/gcp/src/commonMain/kotlin/com/xebia/functional/xef/gcp/GcpChat.kt:107]: Yes, sure! 👍

**Review by Montagon — APPROVED** [review]

**Pull request merged**